src/main/main.ts

```diff
@@ -34,8 +31,33 @@ export function importOnestopToBasecamp(): void {
     deleteOldRows(processedRowIds);
 }
 
+/**
+ * Processes existing event rows by checking whether the row has been modified and if so indiscriminately
+ * updates all the tasks in basecamp with the new information while updating backend metadata
+ * 
+ * If new helper groups are added new tasks in basecamp are created for them
+ * If helper groups are deleted then their corresponding tasks in basecamp will also be deleted.
+ * 
+ * The current state of the row will always be saved into the document properties at the end.
+ * 
+ * @param row 
+ */
 function processExistingRow(row: Row): void {
-    // Will be implemented as part of https://3.basecamp.com/4474129/buckets/38736474/todos/7717428992
+
+    if(hasChanged(row)) {
+
+        const currentRoleRequestMap: RoleRequestMap = getBasecampTodoRequestsForRow(row);
+        const currentRoleTodoIdMap: RoleTodoIdMap  = getRoleTodoIdMap(row);
```

Comment:
small nitpick: i happen to know roleTodoIdMap is what's last saved, but since the name here has "current", if i didn't know it's what's saved, it could be confusing. what do you think of a name like "lastSaved"?

Reply:
That makes sense, I changed the variable names in my latest commit